plugins/ltac/rewrite.ml

```diff
@@ -778,7 +778,8 @@ let resolve_morphism env avoid oldt m ?(fnewt=fun x -> x) args args' (b,cstr) ev
     let morphargs, morphobjs = Array.chop first args in
     let morphargs', morphobjs' = Array.chop first args' in
     let appm = mkApp(m, morphargs) in
-    let appmtype = Typing.unsafe_type_of env (goalevars evars) appm in
+    let sigma, appmtype = Typing.type_of env (goalevars evars) appm in
```

Comment:
In comparison, this one's alright because we want to gather the universe constraints resulting from the newly formed application